lib/modules/manager/flux/extract.ts

```diff
@@ -197,6 +206,22 @@ function resolveResourceManifest(
         deps.push(dep);
         break;
       }
+      case 'OCIRepository': {
+        const dep: PackageDependency<FluxManagerData> = {
+          depName: resource.spec.url?.replace('oci://', ''),
+        };
+        if (resource.spec.ref?.digest) {
+          dep.currentDigest = resource.spec.ref.digest;
+          dep.datasource = DockerDatasource.id;
+        } else if (resource.spec.ref?.tag) {
+          dep.currentValue = resource.spec.ref.tag;
+          dep.datasource = DockerDatasource.id;
+        } else {
+          dep.skipReason = 'unversioned-reference';
+        }
```

Comment:
we should support both too. so renovate can update digest, when the tag points to a new digest, as the docker tags are mutable.

that would be same behavior as we do on other docker based managers

Reply:
I would normally agree but with Flux you can only have one or the other and if both tag and digest are present tag is ignored and digest is used.

Reply:
thats same behavior as docker, digest is preferred.

as aa user i would expect that renovate will follow the tags to update the digest, when both are defined. 

currently renovate would update to digest of latest tag. that would be unexpected by most users.

@rarkins @JamieMagee WDYT?

Reply:
It does appear we can have a digest in the `.spec.ref.tag` field, e.g.

```
v1.8.2@sha256:761c3189c482d0f1f0ad3735ca05c4c398cae201d2169f6645280c7b7b2ce6fc
```

but `.spec.ref.digest` can only have the digest, e.g.

```
sha256:761c3189c482d0f1f0ad3735ca05c4c398cae201d2169f6645280c7b7b2ce6fc
```

Does this help clear things up? If so I can add some tests to verify the former works.

Reply:
Added support for renovating tag and digest in `.spec.tag.ref` with a value `v1.8.2@sha256:761c3189c482d0f1f0ad3735ca05c4c398cae201d2169f6645280c7b7b2ce6fc`

Reply:
Resolving this as it needs a re-review.